Prog/vacant-lots-montgomery_YS.Rmd

```diff
@@ -0,0 +1,1257 @@
+#   filter(rockflag=="1")
+# 
+# vacant_gaithersburg <- vacant1 %>% 
+#   mutate(zoning_code= zoning_bk) %>% 
+#   filter(gaitflag=="1")
+# 
+# zoningdata_rockvile <- left_join(vacant_rockvile, zoning_rockville1, by = "zoning_code")
+# zoningdata_gaithersburg <- left_join(vacant_gaithersburg, zoning_gaithersburg2, by = "zoning_code")
+
+total_rockvile <- jur1 %>% 
+  mutate(X= SHAPE_LEN.x,
+         Y=SHAPE_LEN.y) %>% 
+  mutate(zoning_bk = zoning) %>% 
+  mutate(zoning_code= zoning_bk) %>% 
+  mutate(missing_coord = ifelse(is.na(X) | is.na(Y),
+                                1,
+                                0)) %>% 
+  filter(rockflag=="1")
+
+total_gaithersburg <- jur1 %>% 
+    mutate(X= SHAPE_LEN.x,
+         Y=SHAPE_LEN.y) %>% 
+  mutate(zoning_bk = zoning) %>% 
+  mutate(zoning_code= zoning_bk) %>% 
+  mutate(missing_coord = ifelse(is.na(X) | is.na(Y),
+                                1,
+                                0)) %>% 
+  filter(gaitflag=="1") 
+  
+
+zoningtotal_rockvile <- left_join(total_rockvile, zoning_rockville1, by = "zoning_code")
+zoningtotal_gaithersburg <- left_join(total_gaithersburg, zoning_gaithersburg2, by = "zoning_code")
+```
+
+There are `r ` observations that did not have a match in the zoning code. Below, we see which zoning codes were in the Black Knight data but NOT in the zoning code data.
+
+```{r antijoin}
+# 
+# #Rockville
+# anti_join(vacant_rockvile, zoning_rockville1, by = c("zoning_code")) %>% 
+#   count(zoning_code) %>% 
+#   knitr::kable()
+# 
+# anti_join(total_rockvile, zoning_rockville1, by = c("zoning_code")) %>% 
+#   count(zoning_code) %>% 
+#   knitr::kable()
+# 
+# vacant_rockvile1 <- vacant_rockvile %>% 
+#   mutate(zoning_code= case_when(zoning_code == "R200" ~ "R-200",
+#                                zoning_code == "R60" ~ "R-60",
+#                                zoning_code == "R90" ~ "R-90",
+#                                zoning_code == "R75" ~ "R-75",
+#                                zoning_code == "R150" ~ "R-150",
+#                                TRUE ~ zoning_code))
+# 
+# anti_join(vacant_rockvile1, zoning_rockville1, by = c("zoning_code")) %>% 
+#   count(zoning_code) %>% 
+#   knitr::kable()
+
+total_rockvile1 <- total_rockvile %>% 
+  mutate(zoning_code= case_when(zoning_code == "R200" ~ "R-200",
+                               zoning_code == "R60" ~ "R-60",
+                               zoning_code == "R90" ~ "R-90",
+                               zoning_code == "R75" ~ "R-75",
+                               zoning_code == "R150" ~ "R-150",
+                               TRUE ~ zoning_code))
+
+anti_join(total_rockvile1, zoning_rockville1, by = c("zoning_code")) %>% 
+  count(zoning_code) %>% 
+  knitr::kable()
+
+anti_join(total_gaithersburg, zoning_gaithersburg2, by = c("zoning_code")) %>% 
+  count(zoning_code) %>% 
+  knitr::kable()
+
+#gaithershburg also has R200(5) RE2C(1) which should be a montegomery codes -- what to do? 
```

Comment:
we spatial joined these with the Gaithersburg shapefile so these lots are definitely in Gaithersburg. So I haven't do any calculation on these lots because they don't have zoning parameters merged on.